src/training/training.h

```diff
@@ -77,6 +77,12 @@ class Train : public ModelTask {
     bool restored = !options_->get<bool>("no-restore-corpus")
                     && batchGenerator->restore(trainState);
 
+    // Install custom handler for SIGTERM, to allow for a graceful
+    // shutdown that saves the current state of training before exiting.
+    // This signal handler simply sets a flag that can be checked from
+    // everywhere (getSignalFLAG(SIGTERM); #include common/signal_handling.h)
+    signal(SIGTERM, setSignalFlag);
```

Comment:
Please abstract this together with the `terminationRequested()` function.

Reply:
There is nothing to abstract here. We want to handle a specific case: SIGTERM, not just any termination request. For SIGINT, for example, we want the default behaviour.